### ISSUE TITLE
Dev: WAVE Accessibility: Header (Desktop)

### ISSUE THREAD

**Issue by ryanfchase** [hackforla/tdm-calculator#2531]
### Overview
We need to fix issues detailed by the WAVE Reports that pertain to the Desktop TDM Calculator Header so that we can ensure WCAG compliance across all pages.

### More Info
There is one actionable WAVE Accessibility[^1] error in the desktop implementation of our Header component. This error is triggered when dismissing the login popup dialog while not signed into your account, effectively breaking an expected reference for Assistive Technologies. This happens due to ReactJs-Popup automatically adding the attribute, `aria-describedby`, to the Login link's HTML[^3], yet it fails to remove that attribute when the Popup HTML is removed from the DOM. We fix this by ~manually removing this aria attribute in our on-close event handler.~ replacing `aria-describedby` with `aria-controls`

In a previous version of this ticket, we considered addressing a reported WAVE error for an "empty button" in the header that was hidden with desktop styles. However, this is not necessary: _elements with style_ `display: none` _are hidden from assistive technologies_.[^5]

### Action Items

- [x] ~WAVE Error: Broken ARIA Reference (Login Link + Login Popup Dialog)~
  - [x] ~NavBarLogin.jsx: remove the attribute `aria-describedby`[^2] from `<Popup>`'s trigger element when the modal is closed~
    - [x] ~add id, `popup-trigger` to the inline span defined within the `<Popup>` component~
    - [x] ~modify the `closeModal()` method to remove the `aria-describedby` attribute via `removeAttribute()` [^4]~
- [x] create a react hook that replaces `aria-describedby` with `aria-controls`
  - PR was accidentally skipped, commit: https://github.com/hackforla/tdm-calculator/commit/16072c43a7aa7d18656d133c302190d5a1e53ee9
- [x] undo any extraneous logic introduced by previous PR
  - [x] check to see if this is resolved: https://github.com/hackforla/tdm-calculator/pull/2723#discussion_r2476378260
  - [x] check to see if we need to undo this change: https://github.com/hackforla/tdm-calculator/pull/2558
- [x] smoke test affected components (provide screenshots)
  - [x] ProjectTableRow.jsx
  - [x] NavBarLogin.jsx

#### Related Tickets

- Epic: https://github.com/hackforla/tdm-calculator/issues/2410
  - All WAVE Reports: https://github.com/hackforla/tdm-calculator/issues/1619
  - https://github.com/hackforla/tdm-calculator/issues/2411
  - https://github.com/hackforla/tdm-calculator/issues/2702

#### Staging and Release decks
- [Staging Deck](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg/)
   - ~[Staging slide, 2531](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg/edit?slide=id.g33e1a1d8e1f_1_0#slide=id.g33e1a1d8e1f_1_0)~
- Release Deck
   - [2025-06-10 Release deck, slide ISSUE NUMBER](https://docs.google.com/presentation/d/1s8XuD5jD-LnHMfh2GjfyJsw3w6n_CWl6Rye1uMzxAlw/edit?slide=id.g3608134b6c8_0_570#slide=id.g3608134b6c8_0_570)

#### Screenshots and Excerpts

<details><summary>1. Login Popup</summary>
<p>

## 1.A Login Popup, Screenshot
![Image](https://github.com/user-attachments/assets/2b60b5a1-e88e-4558-a7af-7d02321d411b)

## 1.B Login Popup, Screenshot of WAVE Annotations
- this signifies that this HTML node contains an attribute, `role="tooltip"` (unrelated to the WAVE error in the Action Items)

![Image](https://github.com/user-attachments/assets/ed8ee92f-44f7-4507-9158-317032ed4f1b)

## 1.C Login Popup, Screenshot of HTML
- note the attribute, `id="popup-1"`

![Image](https://github.com/user-attachments/assets/67f6a6bb-21b4-41dc-8995-d88c5ce41908)

---

</p>
</details> 

<details><summary>2. Login Link</summary>
<p>

## 2.A Login Link, Screenshot

![Image](https://github.com/user-attachments/assets/dbf6a9ba-b8c4-4d31-a6e2-7c304cca48ea)

## 2.B Login Link, Screenshot of WAVE Annotations
- Styles were removed to adequately show the WAVE annotation for the Login link
- the WAVE Tool can disrupt styles in order to display the info sidebar, which effectively hid the Login link

![Image](https://github.com/user-attachments/assets/badefdca-387c-436a-b347-ec35c56bfe90)

## 2.C Login Link, Screenshot of HTML
- note the attribute, `aria-describedby="popup-1"`

![Image](https://github.com/user-attachments/assets/14472348-665a-4ec9-acb6-b25eb70ad98f)

## 2.D Broken ARIA Reference, WAVE Tool's Details
> ### What It Means
> An aria-labelledby or aria-describedby reference exists, but the target for the reference does not exist.
> ### Why It Matters
> ARIA labels and descriptions will not be presented if the element referenced does not exist in the page.
> ### How to Fix It
> Ensure the element referenced in the aria-labelledby or aria-describedby attribute value is present within the page and presents a proper label or description.
> ### The Algorithm... in English
> An element has an aria-labelledby or aria-describedby value that does not match the id attribute value of another element in the page.
> ### Standards and Guidelines
> [1.3.1 Info and Relationships (Level A)](https://webaim.org/standards/wcag/checklist#sc1.3.1)
> [4.1.2 Name, Role, Value (Level A)](https://webaim.org/standards/wcag/checklist#sc4.1.2)

---

</p>
</details> 

[^1]: [WAVE Web Accessibility Evaluation Tool: Homepage](https://wave.webaim.org/)
[^2]:  [ARIA Reference: aria-describedby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-describedby)
[^3]: [react-js-popup/src/index.tsx#L213](https://github.com/yjose/reactjs-popup/blob/640f95a287051262297a5d0df3bb82b7ec8d826d/src/index.tsx#L213)
[^4]: [Mdn Web Docs: removeAttribute()](https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute)
[^5]: " Visually hiding elements with `display` or `visibility` hides content from the screen and from assistive technologies."
  [Mdn Web Docs: aria-hidden](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Attributes/aria-hidden)

**Comment by ryanfchase** [comment]
I am re-opening this because, per discussion with @Tony-Villa and @geolunalg, we are still tweaking the fix for this WAVE error.
- agenda: https://github.com/hackforla/tdm-calculator/issues/2410#issuecomment-3560124410
- docs: https://github.com/hackforla/tdm-calculator/issues/2410#issuecomment-3561030939

@Tony-Villa you mentioned you'd continue modifying the hook. When the PR is ready, please use write "Fixes #2531" so that this ticket is also re-closed when the PR closes.

**Comment by Tony-Villa** [comment]
Screenshots for NavBarLogin and ProjectTableRow

<img width="979" height="502" alt="Image" src="https://github.com/user-attachments/assets/5b882616-16b2-45f5-9c72-a299f6b23806" />

<img width="2890" height="1179" alt="Image" src="https://github.com/user-attachments/assets/2edd2b78-9cb8-426f-90db-5cad61bdbced" />

**Comment by ExperimentsInHonesty** [comment]
@Tony-Villa I have to figure out how to fit this into the release deck.  I need to know the PR associated with the commit you referenced

>This issue has been fixed by this commit
https://github.com/hackforla/tdm-calculator/commit/23c3679b44ea1040a0e32bdca71a1faa32d747b0
